app/src/main/java/com/ncnf/storage/FileStore.java

```diff
@@ -0,0 +1,77 @@
+package com.ncnf.storage;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.widget.ImageView;
+
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+import com.ncnf.database.DatabaseResponse;
+
+import java.io.ByteArrayOutputStream;
+import java.util.concurrent.CompletableFuture;
+
+public class FileStore {
+
+    private final long MAX_SIZE = 10 * 1024 * 1024;
+
+    private final FirebaseStorage storage;
+    private final StorageReference fileRef;
+
+    public FileStore(FirebaseStorage storage, String directory, String filename) {
+        this.storage = storage;
+        this.fileRef = storage.getReference().child(directory).child(filename);
+    }
+
+    public FileStore(String directory, String filename) {
+        this.storage = FirebaseStorage.getInstance();
+        this.fileRef = storage.getReference().child(directory).child(filename);
+    }
+
+    public CompletableFuture<DatabaseResponse> uploadImage(Bitmap bitmap) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, baos);
+        byte[] data = baos.toByteArray();
+
+        return upload(data);
+    }
+
+    public CompletableFuture<DatabaseResponse> upload(byte[] data) {
+        CompletableFuture<DatabaseResponse> future = new CompletableFuture<>();
+
+        UploadTask uploadTask = fileRef.putBytes(data);
+        uploadTask.addOnSuccessListener(task -> {
+            future.complete(new DatabaseResponse(true, task.getMetadata(), null));
+        });
+        uploadTask.addOnFailureListener(exception -> {
+            future.complete(new DatabaseResponse(false, null, exception));
+        });
```

Comment:
This part will change with the new DatabaseService since we won't use DatabaseResponse but the CompletableFuture's API which is much cleaner.

Reply:
True! I will wait that we merge the refactor and then update this branch.

Reply:
Okay, thanks a lot !